File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.py
import re

class Solution:
  def myAtoi(self, s: str) -> int:
    s = s.strip()
    if s == '': return 0
    sign = 1
    
    if s[0] == '-':
      sign = -1
    if s[0] in '+-':
      s = s[1:]
    
    digitsRe = "^\d+"
    if not re.search(digitsRe, s): return 0
    
    ans = sign*int(re.match(digitsRe, s)[0])
    return max(ans, -2147483648) if ans < 0 else min(ans, 2147483647)

File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.ts
function myAtoi(s: string): number {
  s = s.trim()
  let sign = 1
  if (s[0] === '-') sign = -1
  if ('+-'.includes(s[0])) s = s.slice(1)
  
  const digitsRe = /^\d+/
  if (!s.match(digitsRe)) return 0
  
  const ans = sign * +s.match(digitsRe)[0]
  return ans < 0 ? Math.max(ans, -2147483648) : Math.min(ans, 2147483647)
}